_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -734,6 +898,38 @@ public static function validate_setup_wizard_questionnaire( $value, $request, $p
 		return true;
 	}
 
+	/**
+	 * Validate the recommendations data
+	 *
+	 * @param array           $value Value to check received by request.
+	 * @param WP_REST_Request $request The request sent to the WP REST API.
+	 * @param string          $param Name of the parameter passed to endpoint holding $value.
+	 *
+	 * @return bool|WP_Error
+	 */
+	public static function validate_recommendations_data( $value, $request, $param ) {
+		if ( ! is_array( $value ) ) {
+			/* translators: Name of a parameter that must be an object */
+			return new WP_Error( 'invalid_param', sprintf( esc_html__( '%s must be an object.', 'jetpack' ), $param ) );
```

Comment:
Did you mean array here, given the check above?

```suggestion
			return new WP_Error( 'invalid_param', sprintf( esc_html__( '%s must be an array.', 'jetpack' ), $param ) );
```

Reply:
For this I did mean object, since objects in JavaScript become arrays in PHP. I figured since the error is returned to the browser it would make sense to phrase it from the point of view of JavaScript.